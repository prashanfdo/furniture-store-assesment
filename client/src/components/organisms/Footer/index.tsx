export default function Footer() {
  return (
    <footer className="px-8 pt-16 pb-12 text-center" data-testid="footer">
      Alle Preise sind in Euro (â‚¬) inkl. gesetzlicher Umsatzsteuer und Versandkosten
    </footer>
  );
}
