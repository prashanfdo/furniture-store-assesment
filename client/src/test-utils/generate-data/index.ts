export * from "./article";
export * from "./category";
export * from "./search";
