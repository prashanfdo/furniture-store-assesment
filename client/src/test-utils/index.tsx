import { render, RenderOptions, renderHook, RenderHookOptions, RenderHookResult } from "@testing-library/react";
import fs from "fs";
import React, { FC, ReactElement } from "react";

import user from "@testing-library/user-event";
import { ProvidersForTests } from "providers";

type GraphQlMocks = React.ComponentProps<typeof ProvidersForTests>["mocks"];

type WrapperProps = React.ComponentProps<typeof ProvidersForTests> & {
  children: React.ReactNode;
};
const Wrapper: FC<WrapperProps> = ({ children, mocks, initialRoute }: WrapperProps) => {
  return (
    <ProvidersForTests mocks={mocks} initialRoute={initialRoute}>
      {children}
    </ProvidersForTests>
  );
};
const wrapper =
  (mocks: GraphQlMocks, initialRoute?: string) =>
  ({ children }: { children: React.ReactNode }) =>
    (
      <Wrapper mocks={mocks} initialRoute={initialRoute}>
        {children}
      </Wrapper>
    );

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper"> & { gqlMocks: GraphQlMocks; initialRoute: string },
) => {
  const view = render(ui, { wrapper: wrapper(options?.gqlMocks, options?.initialRoute), ...options });
  const style = document.createElement("style");
  style.innerHTML = fs.readFileSync("src/index.out.css", "utf8");
  document.head.appendChild(style);

  return view;
};

export const customRenderHook = <T, R>(
  callback: (props: T) => R,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { gqlMocks?: GraphQlMocks; initialRoute?: string },
): RenderHookResult<R, T> => {
  return renderHook(callback, { ...options, wrapper: wrapper(options?.gqlMocks, options?.initialRoute) });
};

export { faker } from "@faker-js/faker";
export * from "@testing-library/react";
export * from "./generate-data";
export { customRender as render };
export { customRenderHook as renderHook };
export { user };
