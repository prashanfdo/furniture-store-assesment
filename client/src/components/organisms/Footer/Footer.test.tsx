import { render, screen } from "test-utils";
import Footer from "./index";

describe("Footer", () => {
  it("should render", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("footer")).toBeInTheDocument();
  });
});
