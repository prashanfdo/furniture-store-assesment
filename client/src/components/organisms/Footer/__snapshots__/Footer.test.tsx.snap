// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
<div>
  <footer
    class="px-8 pt-16 pb-12 text-center"
    data-testid="footer"
  >
    Alle Preise sind in Euro (â‚¬) inkl. gesetzlicher Umsatzsteuer und Versandkosten
  </footer>
</div>
`;
